src/utils.js

```diff
@@ -130,20 +138,58 @@ function setStack(flags, args, clientCommand) {
   }
 
   return stackName
-
 }
 
-function runPulumiCommand(parseInputs, twilioClient, command, commandFlags) {
+async function checkPulumi() {
+  try {
+    const { stdout, stderr } = await exec("pulumi login");
+    if (stderr) {
+      throw new TwilioCliError(stderr);
+    }
+    if (stdout.includes("(file://~)")) {
+      return { local: true };
+    }
+  } catch (err) {
+    throw new TwilioCliError(
+      "Error running Pulumi CLI.\nMake sure the Pulumi CLI is installed\n" + err
+    );
+  }
+}
 
+async function runPulumiCommand(
+  parseInputs,
+  twilioClient,
+  command,
+  commandFlags
+) {
   let { flags, args } = parseInputs;
 
   flags = normalizeFlags(flags);
 
   const stackName = setStack(flags, args, twilioClient);
-  const vars = getEnvironmentVariables(flags, twilioClient, stackName);
-
-  shell.exec(`${vars} ${command} --stack=${stackName} ${commandFlags || ""}`).stdout;
+  let vars = getEnvironmentVariables(flags, twilioClient, stackName);
+
+  const pulumiCLI = await checkPulumi();
+  if (pulumiCLI.local) {
+    const answers = await inquirer.prompt([
+      {
+        type: "password",
+        name: "passPhrase",
```

Comment:
Could this in theory also work if we changed the name to `passphrase` and `answers.passphrase`? I was just wondering why it's using camelcase.

Reply:
Good point @bermudezmt . Done in e1a6f3e